coderd/telemetry/telemetry.go

```diff
@@ -752,6 +764,75 @@ func (r *remoteReporter) createSnapshot() (*Snapshot, error) {
 	return snapshot, nil
 }
 
+func (r *remoteReporter) generateAIBridgeInterceptionsSnapshots(ctx context.Context) ([]AIBridgeInterceptionsSnapshot, error) {
+	// Get the current timeframe, which is the previous hour.
+	now := r.options.Clock.Now().In(time.UTC)
+	endedAtBefore := now.Truncate(time.Hour)
+	startedAtAfter := endedAtBefore.Add(-1 * time.Hour)
+
+	// Note: we don't use a transaction for this function since we do tolerate
+	// some errors, like duplicate heartbeat rows, and we also calculate
+	// snapshots in parallel.
+
+	// Claim the heartbeat row for this hour.
+	err := r.options.Database.InsertTelemetryHeartbeat(ctx, database.InsertTelemetryHeartbeatParams{
+		EventType:          "aibridge_interceptions_snapshot",
+		HeartbeatTimestamp: endedAtBefore,
+	})
+	if database.IsUniqueViolation(err, database.UniqueTelemetryHeartbeatsPkey) {
```

Comment:
We don't need to deduplicate snapshots inside Coder - we can handle that during analysis. Since we know there should only be one snapshot per hour, we can create a view of the data that deduplicates it. I'd just add a sortable unique identifier to the `AIBridgeInterceptionsSnapshot` (e.g. a UUID `id` field) so we can easily differentiate duplicate snapshots.